### ISSUE TITLE
[Bug] Uneven-indent (1007) false positive after the end of a WHILE loop

### ISSUE THREAD

**Issue by tchudobova-cen69317** [MarketSquare/robotframework-robocop#921]
### What happened?

To visually group homogenous keywords in our resource files we usually use comments to name the upcoming keyword section (here the "First section" and the "Second section"). However, we have came across a 'Line is under-indented' warning after the WHILE loop ends, saying that # Second should have the same indent as the END statement.
```
# First section

Example
    ${rc} =   Set Variable    1
    WHILE    ${rc} != 0
        ${rc} =    Keyword that returns zero on success
    END

# Second section

```
The WHILE loop in the above example is pasted from the official Robot Framework User Guide (https://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#while-loops).

![1007_robocop](https://github.com/MarketSquare/robotframework-robocop/assets/130764023/4c630adb-86d0-46dd-ac0c-60c307e7cb81)

### What command/code did you try to run?

The abovementioned code.

### What is the full error message?

Line is under-indentedrobocop[1007](https://robocop.readthedocs.io/en/2.5.0/rules.html#uneven-indent)

### What did you expect to happen instead?

The mentioned comment would not be considered a rule violation.

### Operating System

Windows

### Robocop version

2.5.0 (LSP 1.10.1)
